jerry-core/ecma/base/ecma-helpers-value.c

```diff
@@ -738,7 +738,7 @@ ecma_copy_value (ecma_value_t value)  /**< value description */
     default:
     {
       JERRY_UNREACHABLE ();
-      return ECMA_VALUE_UNDEFINED;
+      break;
```

Comment:
This leads to undefined result.

Reply:
No, it doesn't cause undefined behaviour here either:
- https://github.com/jerryscript-project/jerryscript/blob/9eb9fbf5f12fba7511481d0cabb25a2020d7b710/jerry-core/vm/vm.c#L358
- https://github.com/jerryscript-project/jerryscript/blob/9eb9fbf5f12fba7511481d0cabb25a2020d7b710/jerry-core/ecma/builtin-objects/ecma-builtin-internal-routines-template.inc.h#L211

As I wrote even in the commit message: we should not try to return any fallback value after a failed assert. Assertion error means something disastrous, some design flaw in the program. It's not something we should even try to recover from. (If it's fixable then it's not an assertion.)

Reply:
But still looks ugly. I would recommend merge it with one other option (e.g `case ECMA_TYPE_OBJECT` above), and check with an assertion that the type is really ECMA_TYPE_OBJECT. This technique has already used several places in the code base.